include/SDL3/SDL_stdinc.h

```diff
@@ -47,18 +47,20 @@
 #include <inttypes.h>
 #endif
 #include <stdarg.h>
+
 #ifndef __cplusplus
-#ifdef SDL_DEFINE_STDBOOL
-#ifndef __bool_true_false_are_defined
-#define __bool_true_false_are_defined 1
-#define bool  uint8_t
-#define false 0
-#define true  1
-#endif
-#else
-#include <stdbool.h>
-#endif
+# if defined(SDL_DEFINE_STDBOOL) && defined(__STDC_VERSION__) && __STDC_VERSION__ < 202311L
```

Comment:
This is redundant and doesn't cover the C89 case, where `__STDC_VERSION__` isn't defined. Can you check my PR?